react/README.md

```diff
@@ -297,6 +298,72 @@
   ))}
   ```
 
+## Proptypes
+  - Always use `PropTypes.shape` to define object proptypes. Avoid `PropTypes.object`.
+    ```
+      UserDisplay.propTypes = {
+        user: PropTypes.shape({
+          name: PropTypes.string,
+          email: PropTypes.string,
+          birthday: PropTypes.instanceOf(Date),
+        }),
+      };
+    ```
+  - When a component's proptypes can be used in multiple components, export it as a `const` from the component file.
+    ```
+      export const statusPropTypes = {
+        active: PropTypes.bool.isRequired,
+        message: PropTypes.string.isRequired,
+      };
+      StatusMessage.propTypes = statusPropTypes;
+      export default StatusMessage;
+    ```
+
+    Usage:
+    ```
+      import StatusMessage, { statusPropTypes } from './status-message';
+
+      class AppContainer { };
+      AppContainer.propTypes = {
+        status: statusPropTypes,
+        data: PropTypes.string.isRequired,
+      };
+    ```
+  - When a subset of a component's proptypes can be used in multiple components, export it as a `const` from the component file.
+
+    ```
+      export const partialFilterPropTypes = {
+        searchString: PropTypes.string,
```

Comment:
might add a second prop here (or maybe just a `...`) to make it clear that we wouldn't do this unless there were multiple props